File: RPolishNotation.js
function calc(expr) {
  // TODO: Your awesome code here
  var eachChar=expr.split(" ");
  if(expr==""){
    return 0;
  }
  var stack=new Array(0);
  for(var i=0;i<eachChar.length;i++){
    switch(eachChar[i]){
      case "+":
        var a=stack.pop();
        var b=stack.pop()
        stack.push(a+b);
        break;
      case "-":
        var a=stack.pop();
        var b=stack.pop();
        stack.push(b-a);
        break;
      case "*":
        var a=stack.pop();
        var b=stack.pop();
        stack.push(b*a);
        break;
      case "/":
        var a=stack.pop();
        var b=stack.pop();
        stack.push(b/a);
        break;
      default : stack.push(parseFloat(eachChar[i]));
    }
  }
  return (stack.pop());
}

File: RPolishNotation.java
import java.util.Stack;

public class Calc {

  public double evaluate(String expr) {
    if (expr.equals("")) {
      return 0;
    }
  
    Stack<Double> stack = new Stack<Double>();
    String[] atoms = expr.split(" ");
    
    for (String atom: atoms) {
      Double a, b;
      switch (atom) {
        case "+": stack.push(stack.pop() + stack.pop()); break;
        case "-": b = stack.pop(); a = stack.pop(); stack.push(a - b); break;
        case "*": stack.push(stack.pop() * stack.pop()); break;
        case "/": b = stack.pop(); a = stack.pop(); stack.push(a / b); break;
        default:
          stack.push(Double.parseDouble(atom));
      }
    }
    
    return stack.pop();
  }
}
